TFUtil.py

```diff
@@ -6667,3 +6667,32 @@ def get_positional_encoding(num_channels, length=None, position=None, min_timesc
   signal = tf.concat([tf.sin(scaled_time), tf.cos(scaled_time)], axis=1)
   signal = tf.pad(signal, [[0, 0], [0, num_channels % 2]])  # (length, channels)
   return signal
+
+def check_contains_only_det_ops(print_trace=False):
+  """
+  :param bool print_trace: prints a trace of which ops are checked
+  :return:  are all ops used on the available devices deterministic?
```

Comment:
Make only one space between `:return:` and the following text, not two. (In general, try to be consistent in style with all the remaining code.)
Also better write `:return: whether all ops from the current graph are deterministic`.
Also missing is the return type, i.e. `:rtype: bool`.
Also, better rename the function. `check` does not explain good what the function does. E.g. I expected that it does some `assert` on that or so. If you do not do a check, but you return whether it has some property or not, the function should be called sth like `is_current_graph_deterministic`.

Reply:
Ok, also see the comment in TFUpdater. Instead of just returning a `bool`, we should maybe better return the list of non-deterministic ops. I.e. the function should be called sth like `get_non_deterministic_ops_from_graph`. And then `rtype: list[tf.Operation]` (not just `list[str]` of the op types, it would be good to know where they are exactly, i.e. `tf.Operation`).